apps/meteor/app/api/server/v1/rooms.ts

```diff
@@ -333,6 +334,36 @@ API.v1.addRoute(
 	},
 );
 
+API.v1.addRoute(
+	'rooms.mediaEdit/:rid/:fileId',
+	{ authRequired: true, validateParams: isMediaEditProps },
+	{
+		async post() {
+			if (!(await canAccessRoomIdAsync(this.urlParams.rid, this.userId))) {
+				return API.v1.forbidden();
+			}
+
+			if (!this.bodyParams.fileName) {
```

Comment:
This should be the first check to prevent the database call to check access if fileName is not passed